MetaMask/metamask-extension
# Update `eth-snap-keyring` to 0.2.x to use the `keyring-api` 0.2.x

## Explanation

This PR updates the version of the `eth-snap-keyring` to use the new version of the `keyring-api` that we plan to expose to snap developers.

This PR has no other impacts, and only affects Flask.

**BREAKING CHANGE**: The `keyring-api` 0.2.x is incompatible with the previous 0.1.x release. Snaps developed with version 0.1.x of the API need to migrate to 0.2.x.

## Manual Testing Steps

1. Install the SSK from the [SSK dapp][snap-dapp].
2. Create an account using the [SSK dapp][snap-dapp].
    1. Note that the account name defined in the SSK dapp will be ignored.
3. Use the [test dapp](https://metamask.github.io/test-dapp/) to sign transactions using the sync and async flows.
    1. There is a toggle in the [SSK dapp][snap-dapp] to switch between the flows.

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

[snap-dapp]: https://metamask.github.io/snap-simple-keyring/0.2.2/
